stl/inc/algorithm

```diff
@@ -504,6 +504,48 @@ _NODISCARD _FwdIt adjacent_find(_ExPo&& _Exec, const _FwdIt _First, const _FwdIt
 
 #ifdef __cpp_lib_concepts
 namespace ranges {
+    // VARIABLE ranges::adjacent_find
+    class _Adjacent_find_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <forward_iterator _It, sentinel_for<_It> _Se, class _Pj = identity,
+            indirect_binary_predicate<projected<_It, _Pj>, projected<_It, _Pj>> _Pr = ranges::equal_to>
+        _NODISCARD constexpr _It operator()(_It _First, const _Se _Last, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            // find first satisfying _Pred with successor
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst = _Get_unwrapped(_First);
+            const auto _ULast  = _Get_unwrapped(_Last);
+
+            if (_UFirst != _ULast) {

```

Comment:
Could we turn this into an early return?

Reply:
Yes - I think I'm trying too hard to avoid duplicating the `_Seek_to(_First, meow); return _First;` sequence.